internal/checker/terraform/rules/azure/azurerm_eventgrid_topic_public_network_access_disabled.rego

```diff
@@ -0,0 +1,58 @@
+# METADATA
+# title: "Disable Public Network Access for Azure Event Grid Topic"
+# description: "This policy verifies that public network access is disabled for Azure Event Grid Topics. Disabling public network access enhances security and restricts access to trusted networks."
+# scope: package
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/eventgrid_topic
+# custom:
+#   id: CB_TFAZR_189
+#   severity: MEDIUM
+package lib.terraform.CB_TFAZR_189
+
+import future.keywords.in
+
+supportedResources := ["azurerm_eventgrid_topic"]
```

Comment:
remove this array declaration since it is only 1,
then change is valid with snippet below
isvalid(block) {
	block.Type == "resource"
	"azurerm_eventgrid_topic" in block.Labels
}